Bottom: 3a6e6bad134ceba45a624da7f3012fed4d54f573
Top:    2a9ddfcac427ec14b8ced1791fd0a106156b6812
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-27 16:55:08 -0700

Refresh of rstream-bufsize

---

diff --git a/examples/rstream.c b/examples/rstream.c
index 7fa537e..757a0f6 100644
--- a/examples/rstream.c
+++ b/examples/rstream.c
@@ -350,35 +350,20 @@ out:
 
 static void set_options(int rs)
 {
-	int val, optname, ret;
-	long long bytes;
+	int val, ret;
 	socklen_t size;
 
-	bytes = transfer_size * transfer_count;
-	for (optname = SO_SNDBUF; ; optname = SO_RCVBUF) {
-		size = sizeof val;
-		ret = rs_getsockopt(rs, SOL_SOCKET, optname, (void *) &val, &size);
-		if (ret)
-			break;
-
-		if (val < bytes || buffer_size) {
-			size = sizeof buffer_size;
-			if (!buffer_size)
-				buffer_size = ((val << 2) > bytes) ? bytes : (val << 2);
-			rs_setsockopt(rs, SOL_SOCKET, optname,
-				      (void *) &buffer_size, size);
-		}
-
-		if (optname == SO_RCVBUF)
-			break;
+	if (buffer_size) {
+		size = sizeof buffer_size;
+		rs_setsockopt(rs, SOL_SOCKET, SO_SNDBUF, (void *) &buffer_size, size);
+		rs_setsockopt(rs, SOL_SOCKET, SO_RCVBUF, (void *) &buffer_size, size);
 	}
 
 	val = 1;
 	rs_setsockopt(rs, IPPROTO_TCP, TCP_NODELAY, (void *) &val, sizeof(val));
 
-	if (flags & MSG_DONTWAIT) {
+	if (flags & MSG_DONTWAIT)
 		rs_fcntl(rs, F_SETFL, O_NONBLOCK);
-	}
 }
 
 static int server_connect(void)
